x/emissions/migrations/v4/migrate_test.go

```diff
@@ -62,13 +63,119 @@ func (s *EmissionsV4MigrationTestSuite) SetupTest() {
 		storeService,
 		accountKeeper,
 		bankKeeper,
-		authtypes.FeeCollectorName,
-		keeper.DefaultConfig(),
-	)
+		authtypes.FeeCollectorName)
 
 	s.emissionsKeeper = &emissionsKeeper
 }
 
+// in this test we check that the emissions module params have been migrated
+// and the expected new field is set.
+func (s *EmissionsV4MigrationTestSuite) TestMigrateParams() {
+	storageService := s.emissionsKeeper.GetStorageService()
+	store := runtime.KVStoreAdapter(storageService.OpenKVStore(s.ctx))
+	cdc := s.emissionsKeeper.GetBinaryCodec()
+
+	defaultParams := types.DefaultParams()
+	paramsOld := v3types.Params{
```

Comment:
This should be v2types, isn't it?

Reply:
think "v2types" may need to be renamed "v3types" but i share your question otherwise

Reply:
see my comment on the other conversation.

this is the correct types.

Params needs to be migated from v3 types because v0.4.0 correctly migrated params from 2->3

Topics need to be migrated from the v2 types, because v0.4.0 failed to correctly migrate topics. Hence, the topics need to be migrated from 2 straight to 4.

Reply:
> Topics need to be migrated from the v2 types

Are we sure about this? Because topic types in testnet v0.4.0 are using v3 types too, aren't they?... problem is that one of them didnt get its values changed, and mig to v3 didn't fail because of the nature of the change (additive only) allowed for filling with default values and not break. 

In practice I guess both will work

Reply:
yeah since it takes the value from the GetTopic call, and the only thing that's used is the oldMsg.Id we could _probably_ use the v3 types. But using v2 types is safer because we can only guarantee that the v2 types will be present, we're not guaranteed that the v3 types will be present

probably it's not that important, but v2 is safer so that's what i went with